torch_xla/csrc/tensor_methods.cpp

```diff
@@ -468,6 +470,50 @@ XLATensor XLATensor::_adaptive_avg_pool2d_backward(const XLATensor& grad_output,
       grad_output.GetIrValue(), input.GetIrValue()));
 }
 
+void XLATensor::_amp_foreach_non_finite_check_and_unscale_(
+    std::vector<XLATensor> self, XLATensor& found_inf,
+    const XLATensor& inv_scale) {
+  std::vector<ir::Value> inputs;
+  for (const auto& x : self) {
+    inputs.push_back(x.GetIrValue());
+  }
+  inputs.push_back(found_inf.GetIrValue());
+  inputs.push_back(inv_scale.GetIrValue());
+  absl::Span<ir::Value> inputs_span{inputs};
+  ir::NodePtr node =
+      ir::MakeNode<ir::ops::AmpForachNonFiniteCheckAndUnscale>(inputs_span);
+  for (size_t i = 0; i < self.size(); ++i) {
+    self[i].SetInPlaceIrValue(ir::Value(node, i));
+  }
+  found_inf.SetInPlaceIrValue(ir::Value(node, self.size()));
+}
+
+XLATensor XLATensor::_amp_update_scale(XLATensor growth_tracker,
+                                       const XLATensor& current_scale,
+                                       const XLATensor& found_inf,
+                                       float scale_growth_factor,
+                                       float scale_backoff_factor,
+                                       int growth_interval) {
+  ir::Value scale_growth_factor_ir = GetIrValueForScalar(
+      scale_growth_factor, xla::PrimitiveType::F32, growth_tracker.GetDevice());
+  ir::Value scale_backoff_factor_ir =
+      GetIrValueForScalar(scale_backoff_factor, xla::PrimitiveType::F32,
+                          growth_tracker.GetDevice());
+  ir::Value growth_interval_ir = GetIrValueForScalar(
+      growth_interval, xla::PrimitiveType::S32, growth_tracker.GetDevice());
+  std::vector<ir::Value> inputs;
+  inputs.push_back(growth_tracker.GetIrValue());
+  inputs.push_back(current_scale.GetIrValue());
+  inputs.push_back(found_inf.GetIrValue());
+  inputs.push_back(scale_growth_factor_ir);
+  inputs.push_back(scale_backoff_factor_ir);
+  inputs.push_back(growth_interval_ir);
```

Comment:
ditto, maybe pass these as separate arguments instead of a `vector<ir::Value>`

Reply:
Done.